### ISSUE TITLE
Inconsistencies in bash/shell numeric values

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments-migration-test#951]
(Original issue 1244 created by None on 2016-05-28T16:28:59.626075+00:00)

Pygments version 2.1.3, with this test script:

```
#!bash

myArray=(1 2 3 4 5)

first=1
second=2 # comment
third=3

```

The last number in the array is unhighlighted, unlike the other array numbers. The number before the comment is highlighted, unlike its neighbours.

**Comment by Anteru** [comment]
(Original issue was assigned to tshatch)

(Original issue 1244 last updated on 2016-06-01T07:52:04.505267+00:00)

(Issue automaticaly closed due to status in Bitbucket: resolved)

**Comment by Anteru** [comment]
(Original comment by tshatch on 2016-06-01T07:39:39.487385+00:00)

I have a fix in the works, thanks for the succinct repro.

**Comment by Anteru** [comment]
(Original comment by tshatch on 2016-06-01T07:52:04.498201+00:00)

Improve number highlighting in bash.

Fixes #1244

→ <<cset ce0639f1c9ee>>
